docs/running-on-kubernetes.md

```diff
@@ -340,6 +340,43 @@ RBAC authorization and how to configure Kubernetes service accounts for pods, pl
 [Using RBAC Authorization](https://kubernetes.io/docs/admin/authorization/rbac/) and
 [Configure Service Accounts for Pods](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/).
 
+## Running Spark Thrift Server
+
+Thrift JDBC/ODBC Server (aka Spark Thrift Server or STS) is Spark SQL’s port of Apache Hive’s HiveServer2 that allows
+JDBC/ODBC clients to execute SQL queries over JDBC and ODBC protocols on Apache Spark.
+
+### Client Deployment Mode
+
+To start STS in client mode, excute the following command
+
+```bash
+$ sbin/start-thriftserver.sh \
+    --master k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>
+```
+
+### Cluster Deployment Mode
+
+To start STS in cluster mode, excute the following command
+
+```bash
+$ sbin/start-thriftserver.sh \
+    --master k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port> \
+    --deploy-mode cluster
+```
+
+The most basic workflow is to use the pod name (driver pod name incase of cluster mode and self pod name(pod/container from 
```

Comment:
The script may be run from a client machine outside a k8s cluster. In this case, there's not even a pod. I would suggest separating the explanation of the user flow details by the deploy mode (client vs cluster).

Reply:
Yes, it might be rare to use this type of deployment on k8s, but it's still possible, e.g., for testing, to run the STS in client mode on a client machine. The wording here assumes that the server is running in a pod, which might not be the case.